File: olvi/disclosure_staging/verify_ledger.js
/**
 * OMNIBUS Verification Module - JavaScript/Node.js Context
 * Parses and audits local ledger structure metrics.
 */
const fs = require('fs');
const path = require('path');

const ledgerPath = path.join(__dirname, 'triad_specification.json');

try {
    if (!fs.existsSync(ledgerPath)) {
        console.error(`[-] Error: Ledger file not found at ${ledgerPath}`);
        process.exit(1);
    }

    const rawData = fs.readFileSync(ledgerPath, 'utf8');
    const data = JSON.parse(rawData);
    
    console.log("\n================================================================================");
    console.log("[JS-QUERY] LOCAL COMPLIANCE AUDIT VALIDATION RUN");
    console.log("================================================================================");
    
    let scores = [];
    for (const [layer, details] of Object.entries(data.triad_mapping)) {
        console.log(`\nLayer ID:  ${layer.toUpperCase()}`);
        console.log(`Target:    ${details.component}`);
        console.log(`Vuln:      ${details.vulnerability}`);
        console.log(`CVSS v3.1: [${details.cvss_v31}] - Vector: ${details.vector_string}`);
        scores.push(parseFloat(details.cvss_v31));
    }
    
    const avg = scores.reduce((a, b) => a + b, 0) / scores.length;
    console.log("\n--------------------------------------------------------------------------------");
    console.log(`[+] COMPLIANCE REPORT: Aggregate Core Triad Score: ${avg.toFixed(2)}`);
    console.log("================================================================================\n");

} catch (err) {
    console.error(`[-] Exception running JavaScript validation query: ${err.message}`);
    process.exit(1);
}

File: olvi/disclosure_staging/verify_ledger.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
OMNIBUS Verification Module - Python Engine
Parses and audits local ledger structure metrics.
"""
import os
import json
import sys

def main():
    ledger_path = os.path.join(os.path.dirname(__file__), 'triad_specification.json')
    
    if not os.path.exists(ledger_path):
        print(f"[-] Error: Ledger file not found at {ledger_path}", file=sys.stderr)
        sys.exit(1)
        
    try:
        with open(ledger_path, 'r', encoding='utf-8') as f:
            data = json.load(f)
            
        print("\n================================================================================")
        print("[PYTHON-QUERY] LOCAL COMPLIANCE AUDIT VALIDATION RUN")
        print("================================================================================")
        
        scores = []
        mapping = data.get("triad_mapping", {})
        for layer, details in mapping.items():
            print(f"\nLayer ID:  {layer.upper()}")
            print(f"Target:    {details.get('component')}")
            print(f"Vuln:      {details.get('vulnerability')}")
            print(f"CVSS v3.1: [{details.get('cvss_v31')}] - Vector: {details.get('vector_string')}")
            scores.append(float(details.get('cvss_v31', 0.0)))
            
        if scores:
            avg = sum(scores) / len(scores)
            print("\n--------------------------------------------------------------------------------");
            print(f"[+] COMPLIANCE REPORT: Aggregate Core Triad Score: {avg:.2f}")
            print("================================================================================\n")
            
    except Exception as e:
        print(f"[-] Exception running Python validation query: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == '__main__':
    main()
